var/spack/repos/builtin/packages/py-warpx/package.py

```diff
@@ -36,6 +36,11 @@ class PyWarpx(PythonPackage):
     depends_on('py-wheel', type='build')
     depends_on('warpx +lib ~mpi +shared', type=('build', 'link'), when='~mpi')
     depends_on('warpx +lib +mpi +shared', type=('build', 'link'), when='+mpi')
-
-    def setup_build_environment(self, env):
-        env.set('PYWARPX_LIB_DIR', self.spec['warpx'].prefix.lib)
+    depends_on('cmake@3.15.0:', type='build')
+
+    @property
+    def libs(self):
+        shared = '+shared' in self.spec
+        return find_libraries(
+            ['libwarpx.3d'], root=self.prefix, recursive=True, shared=shared
+        )
```

Comment:
I think this needs to be added to `warpx`, not `py-warpx`. Currently this code doesn't get used because you removed the call to `setup_build_environment`.

Reply:
Yes. Note that the naming of the suffix is: `3d`, `2d` or `rz` depending on the `dims` variant:
https://github.com/ECP-WarpX/WarpX/blob/21.03/setup.py#L77

The `libs` property will only be present if `+libs` is requested.